core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -405,27 +411,25 @@ private UpdateResult shardUpdateOperation(IndexMetaData metaData, BulkShardReque
                 IndexRequest indexRequest = translate.action();
                 try {
                     WriteResult result = shardIndexOperation(bulkShardRequest, indexRequest, metaData, indexShard, false);
-                    return new UpdateResult(translate, indexRequest, result);
-                } catch (Exception e) {
-                    final Throwable cause = ExceptionsHelper.unwrapCause(e);
-                    boolean retry = false;
-                    if (cause instanceof VersionConflictEngineException) {
-                        retry = true;
+                    if (result.success()) {
+                        return new UpdateResult(translate, indexRequest, result);
+                    } else {
+                        return new UpdateResult(translate, indexRequest, isConflictException(result.getFailure()), ExceptionsHelper.unwrapCause(result.getFailure()), null);
                     }
-                    return new UpdateResult(translate, indexRequest, retry, cause, null);
+                } catch (Exception e) {
+                    return new UpdateResult(translate, indexRequest, false, ExceptionsHelper.unwrapCause(e), null);
```

Comment:
Before this would catch parsing errors while indexing, but now we mark them as doc level errors and communicate it through `PrimaryOperationResult`